LayoutTests/accessibility/mixed-contenteditable-visible-character-range-hang.html

```diff
@@ -0,0 +1,113 @@
+<p>
+    <span class="text-style">Line nine line nine</span><span contenteditable="false"> <br role="presentation"/></span>
+    <span class="text-style">Line ten line ten</span><span contenteditable="false"></span>
+</p>
+<p>
+    <span class="text-style">Line eleven line eleven</span><span contenteditable="false"> <br role="presentation"/></span>
+    <span class="text-style">Line twelve line twelve</span><span contenteditable="false"></span>
+</p>
+<p>
+    <span class="text-style">Line thirteen line thirteen</span><span contenteditable="false"> <br role="presentation"/></span>
+    <span class="text-style">Line fourteen line fourteen</span><span contenteditable="false"></span>
+</p>
+<p>
+    <span class="text-style">Line fifteen line fifteen</span><span contenteditable="false"> <br role="presentation"/></span>
+    <span class="text-style">Line sixteen line sixteen</span><span contenteditable="false"></span>
+</p>
+<p>
+    <span class="text-style">Line seventeen line seventeen</span><span contenteditable="false"> <br role="presentation"/></span>
+    <span class="text-style">Line eighteen line eighteen</span><span contenteditable="false"></span>
+</p>
+<p>
+    <span class="text-style">Line nineteen line nineteen</span><span contenteditable="false"> <br role="presentation"/></span>
+    <span class="text-style">Line twenty line twenty </span><span contenteditable="false"></span>
+</p>
+<p>
+    <span class="text-style">Line twentyone line twentyone</span><span contenteditable="false"> <br role="presentation"/></span>
+    <span class="text-style">Line twentytwo line twentytwo</span><span contenteditable="false"></span>
+</p>
+</div>
+
+<button id="button">Button</button>
+
+<script>
+var output = "This test ensures we do not hang when getting the visible character range for a multi-paragraph editable document.\n\n";
+
+if (window.accessibilityController) {
+    window.jsTestIsAsync = true;
+    // Use this to force unobscuredContentRect() to be small (required to trigger the bug). In real VoiceOver usage on
+    // Google Docs and other webpages, sometimes this rect is extremely small (e.g. {x: 0, y: 29, width: 625: height: 1}).
+    testRunner.setViewSize(625, 48);
+    // Offset the y-value from 0 by scrolling (required to trigger the bug, as it makes the contentRect intersect
+    // startingElementRect in AccessibilityObject::visibleCharacterRangeInternal).
+    document.body.scrollTop = 29;
+
+    setTimeout(async function() {
+        const body = accessibilityController.accessibleElementById("body");
+        output += `#body AXVisibleCharacterRange: ${body.stringDescriptionOfAttributeValue("AXVisibleCharacterRange")}\n`;
+
+        // Ensure the main-thread isn't locked up by setting an attribute and checking the result.
+        document.getElementById("button").setAttribute("aria-label", "Foo bar");
+        await waitFor(() => platformTextAlternatives(accessibilityController.accessibleElementById("button")).includes("Foo bar"));
```

Comment:
Is this necessary? I would think that if the main thread were locked, AXVisibleCharacterRange wouldn't return or would return the wrong value making the test fail.

Reply:
You're right, I think this was unnecessary. Removed in the latest commit.